smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/serde/StructuredDataParserGenerator.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.kotlin.codegen.rendering.serde
+
+import software.amazon.smithy.codegen.core.Symbol
+import software.amazon.smithy.kotlin.codegen.rendering.protocol.ProtocolGenerator
+import software.amazon.smithy.model.shapes.MemberShape
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.StructureShape
+
+/**
+ * Responsible for rendering deserialization of structured data (e.g. json, yaml, xml).
+ */
+interface StructuredDataParserGenerator {
```

Comment:
**Question**: Seeing "yaml" in the comment (and humorously contemplating AWS's first YAML-based protocol) led me to a more serious question: do we see a use for these generators beyond protocol serde?

Reply:
No but nothing ties these to a protocol technically